### PR TITLE
added tooltip icon and hover functionality

### LINKED ISSUE #5192: [Feature] Tooltip / Helper on button widget
## Summary

As a developer, I should be able to set a helper text information in the button widget which should show up as a tooltip for the end-user

## Motivation

This will provide additions context about the action for the end-user

## Additional Context

This is a request on [Discord](https://discord.com/channels/725602949748752515/760761686549463060/854961768073920542) from one of our users

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Add tooltip text on Button Widgets

Fixes #5192

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ]  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/btn-widget-tooltip 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 53.38 **(0.03)** | 34.83 **(0.05)** | 31.46 **(0)** | 53.92 **(0.02)**
 :green_circle: | app/client/src/comments/inlineComments/AddCommentInput.tsx | 71.74 **(7.14)** | 20 **(2.3)** | 61.9 **(1.03)** | 72.36 **(6.17)**
 :green_circle: | app/client/src/components/designSystems/blueprint/ButtonComponent.tsx | 48.18 **(2.03)** | 40.66 **(3.01)** | 53.85 **(0)** | 51.52 **(2.06)**</details>

### CODE REVIEW

File: app/client/src/widgets/ButtonWidget.tsx [hunk 1]
```diff
@@ -60,6 +60,16 @@ class ButtonWidget extends BaseWidget<ButtonWidgetProps, ButtonWidgetState> {
             isTriggerProperty: false,
             validation: VALIDATION_TYPES.OPTIONS_DATA,
           },
+          {
+            helpText: "Show help text or details about current button",
```
Comment (somangshu) [inline app/client/src/widgets/ButtonWidget.tsx:64]: Change this to > "Show helper text with button on hover"

File: app/client/src/components/designSystems/blueprint/ButtonComponent.tsx [hunk 2]
```diff
@@ -30,10 +34,27 @@ const getButtonColorStyles = (props: { theme: Theme } & ButtonStyleProps) => {
   }
 };
 
+const ToolTipContent = styled.div`
+  max-width: 350px;
+`;
+
+const ToolTipIcon = styled(IconWrapper)`
```
Comment (somangshu) [inline app/client/src/components/designSystems/blueprint/ButtonComponent.tsx:41]: We dont need a tooltip icon. A hover on anywhere in the button should show the tooltip. Remove all related code

**Review by somangshu — CHANGES_REQUESTED** [review]
Changes as requested

File: app/client/src/components/designSystems/blueprint/ButtonComponent.tsx [hunk 3]
```diff
@@ -369,14 +349,29 @@ function ButtonContainer(
         accent={mapButtonStyleToStyleName(props.buttonStyle)}
         disabled={props.disabled}
         filled={props.buttonStyle !== "SECONDARY_BUTTON"}
-        icon={icon}
+        icon={props.icon}
```
Comment (somangshu) [inline app/client/src/components/designSystems/blueprint/ButtonComponent.tsx]: Why is this required?

**Review by somangshu — CHANGES_REQUESTED** [review]
This is causing a problem to the button skeleton in the loading state. @techbhavin

File: app/client/src/components/designSystems/blueprint/ButtonComponent.tsx (hunk 3 above)
Comment (techbhavin) [reply app/client/src/components/designSystems/blueprint/ButtonComponent.tsx]: @somangshu  here we revert code with same as original code (before we add hover tooltip on the icon but after change request, we change to button hover ) so. you can check with the PR file compare.

Comment (somangshu) [reply app/client/src/components/designSystems/blueprint/ButtonComponent.tsx]: Please check the loading issue otherwise

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
Deploy preview looks good as well

**Comment by YogeshJayaseelan** [comment]
@techbhavin @somangshu - is this applicable for table button type column as well? currently i could see this applicable for button and not for table column button types

**Comment by somangshu** [comment]
This is only for the button widget @YogeshJayaseelan

**Pull request merged**